K8SPSMDB-226 PSMDB database object helm chart (#30)
percona/percona-helm-charts: charts/psmdb-db/README.md

# psmdb-db: A chart for installing Percona Server MongoDB Operator managed Databases

This chart implements Percona Server MongoDB deployment in Kubernets via Custom Resource object. The project itself can be found here:
* <https://github.com/percona/percona-server-mongodb-operator>

## Pre-requisites
* [PSMDB operator](https://hub.helm.sh/charts/percona/psmdb-operator) running in you K8S cluster
* Kubernetes 1.11+
* PV support on the underlying infrastructure - only if you are provisioning persistent volume(s).
* At least `v2.4.0` version of helm

## Custom Resource Details
* <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>

## StatefulSet Details
* <https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/>

## Chart Details
This chart will:

> * deploy a PSMDB database Pods (Custom Resource -> StatefulSet) for the further MongoDB Cluster creation in K8S.

Reviewer: ```suggestion
* deploy PSMDB database Pods (Custom Resource -> StatefulSet) for the further MongoDB Cluster creation in K8S.
```

### Installing the Chart
To install the chart with the `psmdb` release name using a dedicated namespace (recommended):

```sh
helm repo add percona https://percona-lab.github.io/percona-helm-charts/
helm install my-db percona/psmdb-db --version 0.1.0 --namespace my-namespace
```

The chart can be customized using the following configurable parameters:

| Parameter                       | Description                                                                   | Default                                   |
| ------------------------------- | ------------------------------------------------------------------------------| ------------------------------------------|

> | `nameOverride`                  | Set if you the chart name is not desired to be operators name                 | `""`                                      |

Reviewer: ```suggestion
```
It looks like this option is not officially supported

> | `fullnameOverride`              | By default operator name will contain chart name concatenated with operator name.<br>Set this variable if you want to set the whole bunch of names  | `""`                                      |

Reviewer: ```suggestion
```

> | `pause`                         | Stop PSMDB Database safely                                                    | `false`                                   |

Reviewer 2: in Operator - yes, https://github.com/percona/percona-server-mongodb-operator/blob/master/pkg/apis/psmdb/v1/psmdb_defaults.go#L202-L206

| `replset.volumeSpec.pvc.storageClassName`       | ReplicaSet Pods PVC target storageClass                      | `""` |
| `replset.volumeSpec.pvc.accessModes`       | ReplicaSet Pods PVC access policy                      | `[]` |
| `replset.volumeSpec.pvc.resources.requests.storage`       | ReplicaSet Pods PVC storage size                      | `3Gi` |
| |
| `backup.enabled` | Enable backup PBM agent | `true` |
| `backup.restartOnFailure` | Backup Pods restart policy | `true` |
| `backup.image.repository`              | PBM Container image repository                                           | `percona/percona-server-mongodb-operator` |
| `backup.image.tag`                     | PBM Container image tag                                       | `1.4.0-backup`                              |
| `backup.serviceAccountName`   | Run PBM Container under specified K8S SA   | `percona-server-mongodb-operator` |
| `backup.storages`   | Local/remote backup storages settings   | `{}` |
| `backup.tasks`   | Backup working schedule   | `{}` |
| `users`   | PSMDB essential users   | `{}` |

Specify parameters using `--set key=value[,key=value]` argument to `helm install`

## Examples

> ### Deploy a Cluster without a MySQL Proxy, no backups, no persistent disks

Reviewer: ```suggestion
### Deploy a Cluster with no MySQL Proxy, no backups, and no persistent volumes
```
